packages/terra-date-picker/src/DateUtil.js

```diff
@@ -30,6 +30,42 @@ class DateUtil {
     return momentDates.length > 0 ? momentDates : dates;
   }
 
+  // Checks if a given date is out of the range between the start and end dates.
+  static isDateOutOfRange(sourceDate, startDate, endDate) {
+    if (!sourceDate || !sourceDate.isValid() ||
+      !startDate || !startDate.isValid() ||
+      !endDate || !endDate.isValid()) {
+      return false;
+    }
+
+    if (sourceDate.isBefore(startDate) && sourceDate.isAfter(endDate)) {
+      return false;
+    }
+
+    return true;
+  }
+
+  // Checks if a given date is one of the excluded dates.
+  static isDateExcluded(sourceDate, excludedDates) {
+    if (!sourceDate || !sourceDate.isValid()) {
+      return false;
+    }
+
+    const exludeMomentDates = DateUtil.filterInvalidDates(excludedDates);
+
+    if (!exludeMomentDates || exludeMomentDates.length <= 0) {
```

Comment:
Do you need this if check? It looks like your for statement covers the case when this array is empty.

Reply:
Updated: https://github.com/cerner/terra-core/pull/974/commits/f120301502e3d390e72a669fce130604de572ae5